polymc: update to 1.4.0

OriginCode opened the pull request:
Topic Description
-----------------

Update `polymc` to 1.4.0

Package(s) Affected
-------------------

`polymc` 1.4.0

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

OriginCode committed:
polymc: update to 1.4.0

OriginCode commented:
LGTM on amd64:
![圖片](https://user-images.githubusercontent.com/17294071/181114921-2fd08335-1a07-4aa0-8b7d-5f9f51359c13.png)

cthbleachbit commented on extra-games/polymc/autobuild/build:
```diff
@@ -0,0 +1,11 @@
+abinfo "Building polymc ..."
+mkdir -pv "$BLDDIR"
+cd "$BLDDIR"
+cmake "$SRCDIR" -DCMAKE_BUILD_TYPE=Release \
```
Nit: Include `${CMAKE_DEF}` when invoking cmake from custom build scripts.

cthbleachbit replied:
They define `CMAKE_BUILD_TYPE` and `CMAKE_INSTALL_PREFIX` so you don't have to

MingcongBai replied:
> Nit: Include `${CMAKE_DEF}` when invoking cmake from custom build scripts.

Yeah, and `Release` is wrong, you need `RelWithDebInfo`. But you should use CMAKE_DEF, as @cthbleachbit suggests.

OriginCode committed:
polymc: use CMAKE_DEF in `build`

OriginCode commented:
Issue found
![圖片](https://user-images.githubusercontent.com/17294071/181195690-4c4c7950-b2de-4c88-ae2d-b79d9c76fa78.png)

OriginCode committed:
polymc: patch the launcher source code to use sane paths for jars

OriginCode commented:
The issue above is fixed.
![圖片](https://user-images.githubusercontent.com/17294071/181200505-b2c443be-89b6-4877-888b-1e4c1ed8a6a0.png)

OriginCode committed:
multimc5: drop, replace with polymc...

... for redistribution issue
(https://github.com/MultiMC/Launcher#forkingredistributingcustom-builds-policy)
whilst PolyMC is a community fork of MultiMC 5
(https://polymc.org/wiki/overview/faq/#why-did-our-community-choose-to-fork)
which is fully compatible with MultiMC 5.

Signed-off-by: Kaiyang Wu <origincode@aosc.io>

cthbleachbit reviewed (approved):
it werks (TM)
